app/code/community/Bolt/Boltpay/Model/BoltOrder.php

```diff
@@ -345,6 +351,87 @@ protected function getCorrectedTotal($projectedTotal, $magentoDerivedCartData)
+     * If not, and therefore invalid, the address is replaced by using the
+     * provided shipping address.
+     *
+     * @param Mage_Sales_Model_Quote_Address $billingAddress   The billing address
+     *                                                         to be checked
+     * @param Mage_Sales_Model_Quote_Address $fallbackAddress  The address to fallback
+     *                                                         to in case of invalid billing
+     *                                                         address
+     *
+     *
+     * @return bool  true if a correction is made, otherwise false
+     */
+    public function correctBillingAddress(&$billingAddress, $fallbackAddress)
+    {
+        $quote = $billingAddress->getQuote();
+
+        $wasCorrected = false;
+
+        /////////////////////////////////////////////
+        /// Changes persisted to Magento and Bolt
+        /////////////////////////////////////////////
+        if ($fallbackAddress) {
+            if (
+                !trim($billingAddress->getStreetFull()) ||
+                !$billingAddress->getCity() ||
+                !$billingAddress->getCountry()
+            )
+            {
+                $billingAddress
+                    ->setCity($fallbackAddress->getCity())
+                    ->setRegion($fallbackAddress->getRegion())
+                    ->setPostcode($fallbackAddress->getPostcode())
+                    ->setCountryId($fallbackAddress->getCountryId())
+                    ->setStreet($fallbackAddress->getStreet())
+                    ->save();
+
+                $wasCorrected = true;
+            }
+
+            if (!trim($billingAddress->getName())) {
+                $billingAddress
+                    ->setPrefix($fallbackAddress->getPrefix())
+                    ->setFirstname($fallbackAddress->getFirstname())
+                    ->setMiddlename($fallbackAddress->getMiddlename())
+                    ->setLastname($fallbackAddress->getLastname())
+                    ->setSuffix($fallbackAddress->getSuffix())
+                    ->save();
+
+                $wasCorrected = true;
+            }
+        }
+        if ($wasCorrected) $quote->save();
+        /////////////////////////////////////////////
+
+        /////////////////////////////////////////////
+        /// Changes persisted only to Bolt
+        /// These are workarounds that deal with the
+        /// Bolt requirements for fields that are not
+        /// required by Magento and should not be
+        /// persisted in Magento
+        /////////////////////////////////////////////
+        if (
+            !$billingAddress->getRegion() &&
+            !in_array($billingAddress->getCountry(), $this->countriesRequiringRegion)
+        )
+        {
+            $billingAddress->setRegion($billingAddress->getCity());
+            $wasCorrected = true;
+        }
+
+        if (!$billingAddress->getPostcode()) {
+            $billingAddress->setPostcode('-');
```

Comment:
is postal code not required by magento?

Reply:
@daisy1754 
This is a good question.  This varies from store to store depending on configurations.  See these admin pages below:

**Configurable postal code requirement **
![image](https://user-images.githubusercontent.com/973856/50459908-3a437f80-0970-11e9-81a1-d4761975ba78.png)

**Configurable region requirement**
![image](https://user-images.githubusercontent.com/973856/50459939-bc33a880-0970-11e9-91b7-6b0adac94edc.png)